src/chatClient.cpp

```diff
@@ -1337,25 +1441,129 @@ void Client::onUsersUpdate(mega::MegaApi* api, mega::MegaUserList *aUsers)
+                mega::MegaStringMap *userKeyMap;
+                userKeyMap = mega::MegaStringMap::createInstance();
+
+                for (int i = 0; i < sdkPeers->size(); i++)
+                {
+                    //Get peer Handle in B64
+                    mega::MegaHandle peerHandle = sdkPeers->getPeerHandle(i);
+                    char uhB64[12];
+                    mega::Base64::btoa((byte *)&peerHandle, mega::USERHANDLE, uhB64);
+                    uhB64[11] = '\0';
+
+                    //Get peer unified key
+                    auto useruk = kCommand->getKeyByUserId(peerHandle);
+
+                    //Get creator handle in binary
+                    uint64_t auxcreatorHandle = this->myHandle().val;
+                    char *auxcreatorHandleBin = new char[mega::USERHANDLE];
+                    memcpy(auxcreatorHandleBin, &(auxcreatorHandle), mega::USERHANDLE);
+
+                    //Append [creatorhandle+uk]
+                    std::string uKeyBin (auxcreatorHandleBin, mega::USERHANDLE);
+                    uKeyBin.append(useruk->buf(), mega::UNIFIEDKEY);
+
+                    //Encode [creatorhandle+uk] to B64
+                    std::string uKeyB64;
+                    mega::Base64::btoa(uKeyBin, uKeyB64);
+
+                    //Add entry to map
+                    userKeyMap->set(uhB64, uKeyB64.c_str());
+                    delete auxcreatorHandleBin;
+                }
+
+                //Get own Handle in B64
+                uint64_t creatorHandle = this->myHandle().val;
+                char ohB64[12];
+                mega::Base64::btoa((byte *)&creatorHandle, mega::USERHANDLE, ohB64);
+                ohB64[11] = '\0';
+
+                //Get own unified key
+                auto ownKey = kCommand->getKeyByUserId(myHandle());
+
+                //Append [creatorhandle+uk]
+                char *creatorHandleBin = new char[mega::USERHANDLE];
+                memcpy(creatorHandleBin, &(creatorHandle), mega::USERHANDLE);
+                std::string okeyBin (creatorHandleBin, mega::USERHANDLE);
+                okeyBin.append(ownKey->buf(), mega::UNIFIEDKEY);
+
+                //Encode [creatorhandle+uk] to B64
+                std::string oKeyB64;
+                mega::Base64::btoa(okeyBin, oKeyB64);
+
+                //Add entry to map
+                userKeyMap->set(ohB64, oKeyB64.c_str());
+                return api.call(&mega::MegaApi::createPublicChat, sdkPeers.get(), userKeyMap, enctitleB64);
+            });
+        }
+        else
+        {
+            createChatPromise = api.call(&mega::MegaApi::createChat, true, sdkPeers.get(), enctitleB64);
+        }
+
+        return createChatPromise
+        .then([this, wptr](ReqResult result) -> Promise<karere::Id>
+        {
+            if (wptr.deleted())
+                return promise::Error("Chat created successfully, but instance was removed");
+
+            auto& list = *result->getMegaTextChatList();
+            if (list.size() < 1)
```

Comment:
Actually we can enforce `list.size() != 1` here

Reply:
```suggestion
            if (list.size() != 1)
```